### PR TITLE
Hard: repay borrowed coins

### PR DESCRIPTION
**Pull request by denalimarsh**
Implements repay functionality.

Marshaling/setting an empty sdk.Coins object in the store causes a panic which I got around via the changes in `x/harvest/keeper/keeper.go`. Please double-check the approach.

### CODE REVIEW

File: x/harvest/keeper/repay.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/harvest/types"
+)
+
+// Repay borrowed funds
+func (k Keeper) Repay(ctx sdk.Context, sender sdk.AccAddress, coins sdk.Coins) error {
+	// Validate requested repay
```
Comment (karzak) [inline x/harvest/keeper/repay.go:11]: Need to call `k.SyncBorrowInterest` so that repayment includes interest.

**Review by karzak — CHANGES_REQUESTED** [review]
Looking good, few minor things to address.

File: x/harvest/client/cli/tx.go [hunk 2]
```diff
@@ -144,3 +145,31 @@ func getCmdBorrow(cdc *codec.Codec) *cobra.Command {
...
 }
+
+func getCmdRepay(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "repay [1000000000ukava]",
+		Short: "repay tokens to the harvest protocol",
+		Long:  strings.TrimSpace(`repay tokens to the harvest protocol`),
+		Args:  cobra.ExactArgs(1),
+		Example: fmt.Sprintf(
+			`%s tx %s repay 1000000000ukava --from <key>`, version.ClientName, types.ModuleName,
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/client/cli/tx.go]: https://github.com/Kava-Labs/kava/pull/725/commits/3e595a217b5ff5e2d35e00b6f3abf2c80a45e934

File: x/harvest/keeper/repay.go (hunk 1 above)
Comment (denalimarsh) [reply x/harvest/keeper/repay.go:15]: https://github.com/Kava-Labs/kava/pull/725/commits/2723a8e2dbd9722624a007417cd6e30421eb2a91

File: x/harvest/keeper/repay.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+// ValidateRepay validates a requested loan repay
+func (k Keeper) ValidateRepay(ctx sdk.Context, sender sdk.AccAddress, coins sdk.Coins) error {
+	borrow, found := k.GetBorrow(ctx, sender)
+	if !found {
+		return types.ErrBorrowNotFound
+	}
+
+	// TODO: Since interest accumulates every block users will be slightly *underpaying* the outstanding balance
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/keeper/repay.go]: https://github.com/Kava-Labs/kava/pull/725/commits/2723a8e2dbd9722624a007417cd6e30421eb2a91, added test in https://github.com/Kava-Labs/kava/pull/725/commits/1c63777a8adce70bd2210c1f92880e56d673dbde

File: x/harvest/keeper/repay_test.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+				borrowCoins:          sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(50*KAVA_CF))),
+				repayCoins:           sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(60*KAVA_CF))), // Exceeds borrowed coins but not user's balance
+			},
+			errArgs{
+				expectPass: true,
+				contains:   "",
+			},
+		},
+		// {
+		// 	"invalid: insufficent balance for repay",
```
Comment (karzak) [inline x/harvest/keeper/repay_test.go:87]: I would add this test case back in

**Review by karzak — APPROVED** [review]
Changes look good, see comment about including test case for insufficient balance. Otherwise, LGTM.

File: x/harvest/keeper/repay_test.go (hunk 4 above)
Comment (denalimarsh) [reply x/harvest/keeper/repay_test.go]: Whoops, good catch, forgot to uncomment it when updating tests. Fixed in https://github.com/Kava-Labs/kava/pull/725/commits/2a5eb4aac8595c61cf2888529f48eb64a43ad4f5.

**Pull request merged**